project-root/Assets/Scripts/Character.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using InControl;
+
+namespace CharacterScripts
+{
+    public abstract class Character : MonoBehaviour, IDamageable<int, int>, IHealth<int>

```

Comment:
The code pertaining to health in this script was just the framework, and this definitely will not be the final product. Unless it already looks good to you?